sway-core/src/lib.rs

```diff
@@ -265,191 +228,141 @@ pub fn parsed_to_ast(
     errors.extend(new_errors);
     let types_metadata = match types_metadata_result {
         Some(types_metadata) => types_metadata,
-        None => {
-            errors = dedup_unsorted(errors);
-            warnings = dedup_unsorted(warnings);
-            return CompileAstResult::Failure { errors, warnings };
-        }
+        None => return deduped_err(warnings, errors),
     };
 
     // Collect all the types of logged values. These are required when generating the JSON ABI.
-    typed_program.logged_types.extend(
-        types_metadata
-            .iter()
-            .filter_map(|m| match m {
-                TypeMetadata::LoggedType(type_id) => Some(*type_id),
-                _ => None,
-            })
-            .collect::<Vec<_>>(),
-    );
+    typed_program
+        .logged_types
+        .extend(types_metadata.iter().filter_map(|m| match m {
+            TypeMetadata::LoggedType(type_id) => Some(*type_id),
+            _ => None,
+        }));
 
-    let mut cfa_res = perform_control_flow_analysis(&typed_program);
-
-    errors.append(&mut cfa_res.errors);
-    warnings.append(&mut cfa_res.warnings);
-    errors = dedup_unsorted(errors);
-    warnings = dedup_unsorted(warnings);
-    if !errors.is_empty() {
-        return CompileAstResult::Failure { errors, warnings };
-    }
+    // Perform control flow analysis and extend with any errors.
+    let cfa_res = perform_control_flow_analysis(&typed_program);
+    errors.extend(cfa_res.errors);
+    warnings.extend(cfa_res.warnings);
 
     // Evaluate const declarations,
     // to allow storage slots initializion with consts.
     let mut ctx = Context::default();
     let mut md_mgr = MetadataManager::default();
     let module = Module::new(&mut ctx, Kind::Contract);
-    match ir_generation::compile::compile_constants(
+    if let Err(e) = ir_generation::compile::compile_constants(
         &mut ctx,
         &mut md_mgr,
         module,
         &typed_program.root.namespace,
     ) {
-        Ok(()) => (),
-        Err(e) => {
-            errors.push(e);
-            return CompileAstResult::Failure { warnings, errors };
-        }
+        errors.push(e);
     }
 
     // Check that all storage initializers can be evaluated at compile time.
-    let CompileResult {
-        value: typed_program_with_storage_slots_result,
-        warnings: new_warnings,
-        errors: new_errors,
-    } = typed_program.get_typed_program_with_initialized_storage_slots(
+    let typed_wiss_res = typed_program.get_typed_program_with_initialized_storage_slots(
```

Comment:
took me a second to figure out that `wiss` is `with_initialized_storage_slots` 😆